decentraland/unity-renderer
# fix: crash on exit in `explorer-desktop`

## What does this PR change?

When unity is being shut down, objects may be destroyed before the desired flow is completed. 
This PR Prevents multiple errors and exceptions during that phase.

Fixes: decentraland/unity-renderer#1566

## Extra

When the game is being destroyed, for some reason, we are trying to destroy `Transform` components and unity doesn't allow that, throwing an ugly error with no stack trace. So a condition was added to `Utils.SafeDestroy`

## How to test the changes?

Try Desktop build that's using this branch https://github.com/decentraland/explorer-desktop/pull/42
Exit the application and no bug report splash should be seen.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
